documentation/conda.yml

```diff
@@ -0,0 +1,14 @@
+# Python environment and dependencies to build the documentation
+
+name: cocotb-docs
+dependencies:
+  - python=3.8
+  - pip
+# ReadTheDoc dependencies
+  - mock
+  - pillow
+  - sphinx
```

Comment:
Can we pin this to `~= 3` like we do for pip?

Reply:
With conda this syntax does not work.

Reply:
```suggestion
  - sphinx >=3.0,<4
```

Reply:
This also does not work.

Reply:
But it looks pip will take over the requirements in any case. Conda sphinx version is 2.4.4 and if I use this conda.yml I am getting 3.0.2 from pypi (same for ReadTheDoc)

Reply:
And yet it fails if you omit it entirely?

Reply:
>And yet it fails if you omit it entirely?
Yes, It looks like some dependency issue.

Reply:
Unfortunately looks like conda in the yml file supports only "=" so yu can pin particular version only.

Reply:
Claims to be supported here: https://stackoverflow.com/a/54849994/102441

Reply:
What's the failure look like with the syntax I suggested above?

Reply:
So this is maybe that the latest version in conda is 2.4.4 and it fails. I could have missed this.

Reply:
```
ResolvePackageNotFound:
  - sphinx[version='>=3.0,<4']
```

Reply:
Was that all the error there was?

Reply:
You're right though, it does seem conda is on an older version

Reply:
If one skip sphinx one gets this: https://readthedocs.org/projects/cocotb/builds/10910139/

It may be due to the order.  They do this after the user environment:
`conda install --yes --quiet --name 1706 mock pillow sphinx sphinx_rtd_theme`
Why not before?

Reply:
In any case ...

Reply:
> Why not before?

That's what the issue I link to in https://github.com/cocotb/cocotb/pull/1706#issuecomment-619428557 is all about. The flag is the fix for that problem, but they don't want to make it the default in  case it breaks people.

Reply:
Yes, but appending will make no difference practically? Just will not see this in our file.